ca/ca.go

```diff
@@ -342,17 +345,21 @@ func (ca *CAImpl) newCertificate(domains []string, ips []net.IP, key crypto.Publ
 	return newCert, nil
 }
 
-func New(log *log.Logger, db *db.MemoryStore, ocspResponderURL string, alternateRoots int, chainLength int, certificateValidityPeriod uint64) *CAImpl {
+func New(log *log.Logger, clockSource clock.Clock, db *db.MemoryStore, ocspResponderURL string, alternateRoots int, chainLength int, certificateValidityPeriod uint64) *CAImpl {
 	ca := &CAImpl{
 		log:                log,
 		db:                 db,
+		clockSource:        clockSource,
 		certValidityPeriod: defaultValidityPeriod,
 	}
 
 	if ocspResponderURL != "" {
 		ca.ocspResponderURL = ocspResponderURL
 		ca.log.Printf("Setting OCSP responder URL for issued certificates to %q", ca.ocspResponderURL)
 	}
+	if clockSource == nil {
+		ca.clockSource = clock.New()
+	}
```

Comment:
Two comments:
1) Here and throughout, honestly I think it's fine to leave this check/default off. The clk can just be a required argument and pebble can just panic if it isn't provided. Just like it would if the db argument was nil.
2) But if you really do want to keep it optional, do this check/default before the creation of the CAImpl on line 349.